src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +379,284 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+                    if (reader.ReadBlock(buffer, 0, buffer.Length) > 0)
+                    {
+                        byteOffset = encoding.GetByteCount(buffer, 0, buffer.Length);
+                    }
+                }
+            }
+
+            return byteOffset;
+        }
+
+        private int GetRegionByteLength(Region v2Region, Uri uri)
+        {
+            int byteLength = 0;
+            Encoding encoding;
+
+            using (StreamReader reader = GetFileStreamReader(uri, out encoding))
+            {
+                if (reader != null)
+                {
+                    if (v2Region.StartLine > 0) // Use line and column 
+                    {
+                        string sourceLine = null;
+
+                        // Read down to startLine (null return means EOF)
+                        for (int i = 1; i <= v2Region.StartLine && sourceLine != null; sourceLine = reader.ReadLine(), i++) { }
+
+                        if (sourceLine != null && sourceLine.Length <= v2Region.StartColumn)
+                        {
+                            // Since we read past startColumn, we need to back up using the base stream
+                            Stream stream = reader.BaseStream;
+                            stream.Position -= v2Region.StartColumn - 1;
+
+                            // Read the next charLength characters
+                            char[] buffer = new char[v2Region.CharLength];
+                            reader.Read(buffer, 0, buffer.Length);
+
+                            byteLength = encoding.GetByteCount(buffer);
+                        }
+                    }
+                    else // Use charOffset
+                    {
+                        // Read the first charOffset characters
+                        char[] buffer = new char[v2Region.CharOffset];
+                        reader.Read(buffer, 0, buffer.Length);
+
+                        // Read the next charLength characters  
+                        buffer = new char[v2Region.CharLength];
+                        reader.Read(buffer, 0, buffer.Length);
+
+                        byteLength = encoding.GetByteCount(buffer);
+                    }
+                }
+            }
+
+            return byteLength;
+        }
+
+        private int GetRegionEndColumn(Region v2Region, Uri uri)
+        {
+            int endColumn = 0;
+            Encoding encoding;
+
+            using (StreamReader reader = GetFileStreamReader(uri, out encoding))
+            {
+                if (reader != null)
+                {
+                    string sourceLine = null;
+
+                    // Read down to endLine (null return means EOF)
+                    for (int i = 1; i <= v2Region.EndLine && sourceLine != null; sourceLine = reader.ReadLine(), i++) { }
+
+                    if (sourceLine != null && sourceLine.Length < v2Region.EndColumn)
+                    {
+                        endColumn = sourceLine.Length;
```

Comment:
Bug: Must be `sourceLine.Length + 1`, because regions are "half-open": That is, they include the start column but not the end column. So if the line is 5 characters long, and you want to specify the entire line, you must set endColumn to 6.

Reply:
Right, fixed